### PR TITLE
[FIX]: Attempt creation of symlink on Windows and handle exceptions.

### PR DESCRIPTION
**Pull request by slowglow**
## Description
If symlinks are not enabled on Windows, issue a guiding message.

## Motivation and Context
See https://github.com/glotzerlab/signac/issues/430#issuecomment-1809706816

## Checklist:
- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac/blob/main/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac/blob/main/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/main/contributors.yaml).
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac/tree/main/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/main/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: signac/linked_view.py [hunk 1]
```diff
@@ -85,8 +79,29 @@ def create_linked_view(project, prefix=None, job_ids=None, path=None):
...
-    _check_directory_structure_validity(links.keys())
-    _update_view(prefix, links)
+
+    # Updating the view will fail on Windows, if symlinks are not enabled.
+    # Before re-raising the exception, print a helpful message for the expected error.
+    try:
+        _check_directory_structure_validity(links.keys())
+        _update_view(prefix, links)
+    except OSError as err:
+        if (sys.platform == "win32") & (err.winerror == 1314):
```
Comment (bdice) [inline signac/linked_view.py:89]:
```suggestion
        if sys.platform == "win32" and err.winerror == 1314:
```

File: signac/linked_view.py [hunk 2]
```diff
@@ -85,8 +79,29 @@ def create_linked_view(project, prefix=None, job_ids=None, path=None):
...
+            print("To enable the creation of symlinks on Windows you need")
+            print("to enable 'Developer mode' (requires administrative rights).")
+            print(" ")
+            print("To enable 'Developer mode':")
+            print(" 1. Go to 'Settings'.")
+            print(" 2. In the search bar type 'Use developer features'")
+            print(" 3. Enable the item 'Developer mode'.")
+            print("The details for Home edition and between Windows versions may vary.")
+            print("-----------------------------------------------------------------")
+            print(" ")
```
Comment (bdice) [inline signac/linked_view.py:103]: I'm going to rewrite this block with `textwrap.dedent` and a single string block.

**Review by bdice — APPROVED** [review]
I applied some changes to the tests. I verified locally that these tests are skipped on Windows when Developer Mode is disabled, and they pass on Windows when Developer Mode is enabled. This looks good to me.

@b-butler @cbkerr @joaander if you have feedback, feel free to weigh in -- otherwise I think this is good to merge.

**Review by joaander — APPROVED** [review]
Thanks @slowglow and @bdice! The change is reasonable to allow functionality previously not allowed and the unit tests are updated appropriately.

**Comment by bdice** [comment]
FYI, CI looks blocked (perhaps due to #957) because the expected branches are changing.

image

**Comment by joaander** [comment]
> FYI, CI looks blocked (perhaps due to #957) because the expected checks are changing.

I removed the 3.12 required checks for now. #957 appears to have some issues with pandas that may take some time to resolve.

**Review by cbkerr — APPROVED** [review]
Thank you!

**Pull request merged**